pkg/operator/ceph/webhook-config.go

```diff
@@ -182,91 +189,14 @@ func createValidatingWebhookConfig(ctx context.Context, clusterdContext *cluster
 				"cert-manager.io/inject-ca-from": fmt.Sprintf("%s/rook-admission-controller-cert", namespace),
 			},
 		},
-		Webhooks: []admv1.ValidatingWebhook{
-			{
-				Name: fmt.Sprintf("cephcluster-wh-%s-%s.rook.io", admissionControllerAppName, namespace),
-				Rules: []admv1.RuleWithOperations{
-					{
-						Rule: admv1.Rule{
-							APIGroups:   []string{"ceph.rook.io"},
-							APIVersions: []string{"v1"},
-							Resources:   []string{"cephclusters"},
-						},
-						Operations: []admv1.OperationType{
-							admv1.Update,
-							admv1.Create,
-							admv1.Delete,
-						},
-					},
-				},
-				ClientConfig: admv1.WebhookClientConfig{
-					Service: &admv1.ServiceReference{
-						Name:      admissionControllerAppName,
-						Namespace: namespace,
-						Path:      &serviceCephClusterPath,
-					},
-				},
-				AdmissionReviewVersions: []string{"v1", "v1beta1"},
-				SideEffects:             &sideEffects,
-				TimeoutSeconds:          &timeout,
-			},
-			{
-				Name: fmt.Sprintf("cephblockpool-wh-%s-%s.rook.io", admissionControllerAppName, namespace),
-				Rules: []admv1.RuleWithOperations{
-					{
-						Rule: admv1.Rule{
-							APIGroups:   []string{"ceph.rook.io"},
-							APIVersions: []string{"v1"},
-							Resources:   []string{"cephblockpools"},
-						},
-						Operations: []admv1.OperationType{
-							admv1.Update,
-							admv1.Create,
-							admv1.Delete,
-						},
-					},
-				},
-				ClientConfig: admv1.WebhookClientConfig{
-					Service: &admv1.ServiceReference{
-						Name:      admissionControllerAppName,
-						Namespace: namespace,
-						Path:      &serviceCephBlockPoolPath,
-					},
-				},
-				AdmissionReviewVersions: []string{"v1", "v1beta1"},
-				SideEffects:             &sideEffects,
-				TimeoutSeconds:          &timeout,
-			},
-			{
-				Name: fmt.Sprintf("cephobjectstore-wh-%s-%s.rook.io", admissionControllerAppName, namespace),
-				Rules: []admv1.RuleWithOperations{
-					{
-						Rule: admv1.Rule{
-							APIGroups:   []string{"ceph.rook.io"},
-							APIVersions: []string{"v1"},
-							Resources:   []string{"cephobjectstores"},
-						},
-						Operations: []admv1.OperationType{
-							admv1.Update,
-							admv1.Create,
-							admv1.Delete,
-						},
-					},
-				},
-				ClientConfig: admv1.WebhookClientConfig{
-					Service: &admv1.ServiceReference{
-						Name:      admissionControllerAppName,
-						Namespace: namespace,
-						Path:      &serviceCephObjectStorePath,
-					},
-				},
-				AdmissionReviewVersions: []string{"v1", "v1beta1"},
-				SideEffects:             &sideEffects,
-				TimeoutSeconds:          &timeout,
-			},
-		},
 	}
 
+	for i := range resources {
+		validatingWebhook.Webhooks = append(validatingWebhook.Webhooks, webhookRules(resourcesWebhookName[i], resources[i], resourcesServicePath[i]))
+	}
+
+	validatingWebhook.Webhooks = append(validatingWebhook.Webhooks, webhookRules(fmt.Sprintf("cephcluster-wh-%s-%s.rook.io", admissionControllerAppName, namespace), "", ""))
```

Comment:
Is this a duplicate from the other entry for cephcluster in the slice above, or what is different about it? And why not include it in the slice above?

Reply:
actually it should be inside the loop, leftover from testing